### PR TITLE
Forward WebSocket requests

### PR DESCRIPTION
**Pull request by shosti**
This allows mitm to properly handle WebSocket requests, which previously
led to a wide variety of errors. WebSocket requests just skip most of the
proxy machinery and directly forward the connection.

Relates to https://github.com/rainforestapp/yankee/issues/40.

@rainforestapp/core @ukd1

### CODE REVIEW

File: mitm.go [hunk 1]
```diff
@@ -119,15 +120,20 @@ var (
 )
 
 func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if p.Director == nil {
```
Comment (nxvl) [inline mitm.go:123]: Do we need this to be a function "global"? wouldn't make more sense inside the isWebSocket block?

File: mitm.go [hunk 2]
```diff
@@ -173,6 +179,57 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
...
+		return
+	}
+	nc, _, err := w.(http.Hijacker).Hijack()
+	if err != nil {
+		log.Printf("forwardWebSocket: hijack error: %v", err)
+		http.Error(w, "No Upstream", 503)
+		return
+	}
+	defer nc.Close()
+	defer d.Close()
```
Comment (nxvl) [inline mitm.go]: Can you please get the defers closer to the variable instantiation. If something goes terribly wrong on Hijack(), the dial won't get closed.

**Comment by nxvl** [comment]
Other than 2 minor things, :ship:

File: mitm.go (hunk 2 above)
Comment (shosti) [inline mitm.go]: Good catch, will change

File: mitm.go (hunk 1 above)
Comment (shosti) [inline mitm.go:123]: This is also necessary for `ReverseProxy` so it makes sense to be at the beginning (it already happened, just moved it up).

**Comment by shosti** [comment]
@nxvl updated, :+1: :question:

File: mitm.go (hunk 1 above)
Comment (nxvl) [inline mitm.go:123]: kk

**Pull request merged**